src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -1138,6 +1138,21 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    .rs.scalarListFromList(defaultOptions)
 })
 
+.rs.addFunction("executeChunkCallback", function(chunkName, chunkCode)
+{
+  if (exists(".rs.notebookChunkCallbacks"))
+  {
+     results <- list()
+     handles <- ls(.rs.notebookChunkCallbacks)
+     for (handle in handles) {
+        chunkCallback <- get(handle, envir = .rs.notebookChunkCallbacks)
+        results <- c(results, chunkCallback(chunkName, chunkCode))
```

Comment:
This should be wrapped in a try/catch so we can tolerate failures occurring in the user-supplied `chunkCallback` function.